### PR TITLE
virt-api: dialer should wrap ipv6 address with square brackets to portfoward ipv6 VM

### LINKED ISSUE #9850: virt-api: can't portforward for IPv6 VM
**What happened**:
For IPv6 VM, virt-api port-forward handler not works, which cause `virtctl ssh` command throw error.
```
# ./virtctl ssh v3
can't access VMI v3: Internal error occurred: dialing VM: dial tcp: address 5408:4003:10bb:6a01:83b9:6360:c66d:1a:22: too many colons in address
```

My kubernetes cluster is deploy as IPv6-first, so each pod has a IPv6 address.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Steps to reproduce the behavior.

**Additional context**:

**Environment**:
- KubeVirt version (use `virtctl version`): 0.59.0

### PR DESCRIPTION
**Pull request by lxs137**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #9850 

**Special notes for your reviewer**:

**Release note**:

```release-note
virt-api: portfowrad can handle IPv6 VM
```

### CODE REVIEW

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @lxs137.

Could you please update the commit message to include more information about the problem this PR solves?

**Comment by lxs137** [comment]
> Thank you for the PR @lxs137.
> 
> Could you please update the commit message to include more information about the problem this PR solves?

@orelmisan I have updated the commit message :-)

File: pkg/virt-api/rest/dialers.go [hunk 1]
```diff
@@ -68,7 +69,12 @@ func (n netDial) DialUnderlying(vmi *v1.VirtualMachineInstance) (net.Conn, *erro
 		protocol = protocolParam
 	}
 
-	addr := fmt.Sprintf("%s:%s", targetIP, port)
+	var addr string
+	if netutils.IsIPv6String(targetIP) {
+		addr = fmt.Sprintf("[%s]:%s", targetIP, port)
+	} else {
+		addr = fmt.Sprintf("%s:%s", targetIP, port)
+	}
```
Comment (maiqueb) [inline pkg/virt-api/rest/dialers.go:77]: I'd default to ipv4:

```golang
        addr := fmt.Sprintf("%s:%s", targetIP, port)
	if netutils.IsIPv6String(targetIP) {
		addr = fmt.Sprintf("[%s]:%s", targetIP, port)
	}
```

**Review by maiqueb — CHANGES_REQUESTED** [review]
Can you add a unit test for this ? 

I honestly haven't looked at the surrounding code, thus I am unaware of how feasible / hard it is to create one.

**Comment by orelmisan** [comment]
@lxs137 In the network e2e tests there is a [comment](https://github.com/kubevirt/kubevirt/blob/2426905e43fa74ebbef528c456aa65107a926e73/tests/network/port_forward.go#L49) about port forwarding with IPv6 not being currently supported, with a link to a tracking [issue](https://github.com/kubevirt/kubevirt/issues/7276). 

I've asked @AlonaKaplan (the network maintainer) to take a look at this PR.

**Review by maiqueb — APPROVED** [review]
Perfect! 

Thanks for adding these neat unit tests. 

Nice contribution.

**Pull request merged**